src/app/icd/ICDManager.cpp

```diff
@@ -488,6 +491,17 @@ void ICDManager::OnICDManagementServerEvent(ICDManagementEvents event)
     }
 }
 
+void ICDManager::OnSubscriptionReport()
+{
+    assertChipStackLockedByCurrentThread();
```

Comment:
fwiw `UpdateOperationState` already as this check and is the next call.

Reply:
removed the check